File: 0930-binary-subarrays-with-sum/0930-binary-subarrays-with-sum.py
class Solution:
    def numSubarraysWithSum(self, nums: List[int], goal: int) -> int:


        mp = {}

        total = 0

        cnt = 0


        for i in range(len(nums)):

            total+=nums[i]

            if total==goal:
                cnt+=1

            
            if mp.get(total-goal, 0):
                cnt+=mp.get(total-goal,0)
                       
            mp[total] = mp.get(total , 0) + 1


            
        return cnt
            
           

                
        

File: 0930-binary-subarrays-with-sum/0930-binary-subarrays-with-sum.js
/**
 * @param {number[]} nums
 * @param {number} goal
 * @return {number}
 */
var numSubarraysWithSum = function(nums, goal) {


    let number = 0

    for(let i=0 ; i<nums.length ; i++){
        let sum = 0;
         for (let j = i ; j<nums.length; j++){

            sum +=nums[j];
            if(sum=== goal) number++
             
         }
    }


    return number;
    
};